src/sdk-installer.ts

```diff
@@ -14,7 +16,32 @@ const CMDLINE_TOOLS_URL_LINUX = 'https://dl.google.com/android/repository/comman
 export async function installAndroidSdk(apiLevel: number, target: string, arch: string, emulatorBuild?: string, ndkVersion?: string, cmakeVersion?: string): Promise<void> {
   const isOnMac = process.platform === 'darwin';
 
-  if (!isOnMac) {
+  // Check if ANDROID_HOME is set.
+  if (!process.env.ANDROID_HOME) {
+    core.setFailed('ANDROID_HOME is a required environment variable and is not set.\nPlease double check your host settings.');
+    return;
+  }
+
+  if (fs.existsSync(process.env.ANDROID_HOME)) {
+    console.log('Using previous installation of base Android SDK, found on ${process.env.ANDROID_HOME}');
+  } else {
+    const installed = await installBaseSdk();
+    if (!installed) {
+      core.setFailed('Could not install base Android SDK.');
+      return;
+    }
+    const licenses = await acceptLicenses();
+    if (!installed || !licenses) {
+      core.setFailed('Could not accept Android SDK licenses.');
+      return;
+    }
+  }
+
+  // self-hosted
+  const selfHosted = core.getInput('self-hosted');
+  // It is not required to configure permissions on self-hosted and macos
+  // environment.
+  if (!isOnMac && !selfHosted) {
     await exec.exec(`sh -c \\"sudo chown $USER:$USER ${process.env.ANDROID_HOME} -R`);
```

Comment:
So this is only required for non-self-hosted Linux VMs. Maybe we can just do a fresh installation when host is linux to simplify things? Running the action on `ubuntu` VM is already too slow anyway (I want to look at supporting windows instead which also has HAXM installed).